modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/OpenAPIV3Parser.java

```diff
@@ -37,161 +37,96 @@
     public SwaggerParseResult readLocation(String url, List<AuthorizationValue> auth, ParseOptions options) {
         SwaggerParseResult result = new SwaggerParseResult();
         try {
-
-            result = readWithInfo(url,auth);
-
-            if (result.getOpenAPI() != null) {
-                String version = result.getOpenAPI().getOpenapi();
-                if (auth == null) {
-                    auth = new ArrayList<>();
-                }
-                if (version != null && version.startsWith("3.0")) {
-                    if (options != null) {
-                        OpenAPIResolver resolver = new OpenAPIResolver(result.getOpenAPI(), auth, null);
-                        if (options.isResolve()) {
-                            result.setOpenAPI(resolver.resolve());
-                        }
-                        if (options.isResolveFully()) {
-                            result.setOpenAPI(resolver.resolve());
-                            new ResolverFully().resolveFully(result.getOpenAPI());
-                        }
-                    }
-                }
-            }
-
+            String data = getDataFromLocation(url, auth);
+            result = readContents(data, auth, options);
+        }
+        catch (SSLHandshakeException e) {
+            result.setMessages(Arrays.asList("unable to read location `" + url + "` due to a SSL configuration error.  " +
+                  "It is possible that the server SSL certificate is invalid, self-signed, or has an untrusted " +
+                  "Certificate Authority."));
         }
-
         catch (Exception e) {
-            result.setMessages(Arrays.asList(e.getMessage()));
+            result.setMessages(Arrays.asList("unable to read location `" + url + "`"));
         }
         return result;
     }
 
-    public OpenAPI read(String location) {
```

Comment:
Hi, Why remove this method?

Reply:
Hi,
IMHO this method is not useful. `readContents` and `readLocation` are enough. But if `read(location)` has to be maintened, maybe should we call it `readAndResolve` or `readWithResolve` ?

And about the other `read` method, it is a duplicate of the `io.swagger.parser.OpenAPIParser`. This class is already a `SwaggerExtension`, to my mind, it doesn't have to know the others existing extensions (And maybe I should have remove `getExtensions` and `transform` too in my PR)

Finally, we could just have :
```java
public SwaggerParseResult readLocation(String url, List<AuthorizationValue> auth, ParseOptions options) {...}
public SwaggerParseResult readContents(String swaggerAsString, List<AuthorizationValue> auth, ParseOptions options) {...}

private String getDataFromLocation(String location, List<AuthorizationValue> auths) throws Exception {...}
private JsonNode mapDataToJsonNode(String data) throws JsonProcessingException, IOException {...}
private SwaggerParseResult readWithInfo(JsonNode node){...}

//and for instance
public OpenAPI readAndResolve(String location) {
    ParseOptions options = new ParseOptions();
    options.setResolve(true);
    return readLocation(location, null, options).getOpenAPI();
}
```

But this is just my opinion :smile: